include/eve/arch/cpu/wide.hpp

```diff
@@ -401,9 +401,9 @@ namespace eve
     //! @brief Performs a compound bitwise and on all the wide lanes and assign the result to the current
     //! one
     template<value V>
-    friend EVE_FORCEINLINE auto operator&=(wide& w, V o) noexcept
+    friend EVE_FORCEINLINE wide& operator&=(wide& w, V o) noexcept
 #if !defined(EVE_DOXYGEN_INVOKED)
-        requires(!kumi::product_type<Type> && bit_compatible_values<wide, V>)
+        requires (!kumi::product_type<Type> && supports_bitwise_call<wide, V>)
```

Comment:
the word "supports bitwise call" is not really sayinmg much too. It would be fine if that had a doc but it doesn't.

What are the actual requirements? same size?
Usually we cast scalars to Type - that doesn't work here?

Can we bit and doubles? C++ standar can't. https://godbolt.org/z/hhfvYM7Kh

I appreciate that you didn;t do this originally, but let's fix it up proper, since you touched it

Reply:
This is kinda not awesome.

1. I'm ify about bit operations on floats, since the standard doesn't do it.
2. two scalars - the same size in bits - doesn't work, this operator will never be called for scalars.
3. "element of the wide and the scalar have the same size" - this is inconistent with how other operators work, I believe we always cast. 

@jfalcou 

How about

```
wide& operator&=(Type x); // sclaras are implicitly cast, only integrals

wide& operator&=(wide<U, N>) // both are integrals. same element size. 
```

Reply:
yeah - the problem is there.

ok then - how about `same_element_size_value` concept? and not product?

Reply:
Operator& on wide won't let us do `float & float` anyway.

The semantic of bit_and(x,y) is to treat y as a bitmask and return x & bits(y) in the same type of x. 
Has been that since ages and it works.

The semantic of bit operations taking whatever as long as the # of bits fits is quite the source of a lot of optimisation in EVE. This semantic is documented properly for bit_* functions and changing all the place where we use it means 100s of bit_cast noisy changes. So I am not too keen on that.

Reply:
I deeply agree with joel point of view. I am fiercely (but kindly) against changes to that semantic.